sdk/core/azure-core/azure/core/pipeline/transport/requests_asyncio.py

```diff
@@ -102,6 +102,14 @@ class AsyncioRequestsTransport(RequestsTransport, AsyncHttpTransport):  # type:
         loop = kwargs.get("loop", _get_running_loop())
         response = None
         error = None # type: Optional[Union[ServiceRequestError, ServiceResponseError]]
+        if hasattr(request.data, '__aiter__'):
+            # Need to consume that async generator, since requests can't do anything with it
+            # That's not ideal, but a list is our only choice. Memory not optimal here,
+            # but providing an async generator to a requests based transport is not optimal too
+            new_data = []
+            async for part in request.data:
+                new_data.append(part)
+            request.data = iter(new_data)
```

Comment:
It looks like the subsequent use pull individual attributes off `request` instead of passing the request object itself, so is it necessary to mutate `request` here? (Might be unexpected side effect to callers?)

Reply:
This case also seems worth putting under test, if feasible

Reply:
sounds good